### PR TITLE
Remove private sessions #8447

### LINKED ISSUE #8447: Remove private sessions
**Description of feature/enhancement:** As according to the discussion in https://github.com/TEAMMATES/teammates/pull/8408#issuecomment-364850538, we should remove the `Private` session feature.

**Justification:**

As pointed out by @damithc sir, this feature is not widely used. Apart from that, `Private` sessions also give rise to issues like #8406, #8411.

**Original Discussion :** https://github.com/TEAMMATES/teammates/pull/8408#issuecomment-364770482

### PR DESCRIPTION
**Pull request by sukanta-27**
Fixes #8447 

**Outline of Solution:**
- [x] Remove `Never` option from feedback form UI 
- [x] Remove private session logic from main Logic  
- [x] Remove `private` session testcases from test
- [x] Remove `private` sessioin JSON data
- [x] Update test pages with GodMode
- [x] Update Instructor Help page and necessary documentations (if any)
- [x] Remove `FeedbackSessionType` attribute
- [x] Rename `feedbackSessionType` to `sessionTemplateType`

**Feedback Session form:**
**UI before:**
screenshot 358
**UI after:**
screenshot 380
**Feedback Session Table:**
**UI before:**
screenshot 378
**UI after:**
screenshot 379

Situations like the first three sessions where there the start time and/or end time is null, will be solved after data migration. (**Discussion:** https://github.com/TEAMMATES/teammates/issues/8447#issuecomment-367753028)

**Status: Ongoing**

### CODE REVIEW

File: src/test/java/teammates/test/cases/search/FeedbackResponseCommentSearchTest.java [hunk 1]
```diff
@@ -25,8 +25,9 @@ public void allTests() {
                 .get("comment1FromT1C1ToR1Q1S1C1");
         FeedbackResponseCommentAttributes frc1I1Q2S1C1 = dataBundle.feedbackResponseComments
                 .get("comment1FromT1C1ToR1Q2S1C1");
-        FeedbackResponseCommentAttributes frc1I3Q1S1C2 = dataBundle.feedbackResponseComments
-                .get("comment1FromT1C1ToR1Q1S1C2");
+        //Modify it after discussing, see if the build passes
+        /*FeedbackResponseCommentAttributes frc1I3Q1S1C2 = dataBundle.feedbackResponseComments
```
Comment (sukanta-27) [inline src/test/java/teammates/test/cases/search/FeedbackResponseCommentSearchTest.java]: @whipermr5 Want to clarify something. The tests that I have commented out are dealing with the `Private` session that is in `typicalDataBundle.json` file. But the tests are not `Private session` related itself. Should I keep them or should I remove them, as the private session data in `typicalDataBundle.json` has already been deleted.

Comment (sukanta-27) [reply src/test/java/teammates/test/cases/search/FeedbackResponseCommentSearchTest.java]: @whipermr5 Used the other session in that course `Instructor feedback session` instead, modified the private session's question,comment,response slightly changed it to `Instructor feedback session`'s questioin,comment and response.

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 2]
```diff
@@ -2258,6 +2181,7 @@ public boolean isFeedbackSessionFullyCompletedByStudent(
         return true;
     }
 
+    @SuppressWarnings("PMD.UnusedPrivateMethod")
     private boolean isFeedbackSessionFullyCompletedByInstructor(
```
Comment (sukanta-27) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: @whipermr5 This method is not currently in use, as it was used in `Case PRIVATE:`, but this seems to be a useful method, that we can use later for other purposes (Maybe), which is why I did not remove it. Let me know if you decide otherwise.

Comment (wkurniawan07) [reply src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: Don't let "we can use later for other purposes" be a reason against removing dead code, unless that "later" is very immediate. "This class/method/... looks useful, let's save it for a rainy day" is very often the main reason for accumulating dead code.

Comment (sukanta-27) [reply src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: Okay. Removing it. :sweat_smile:

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java [hunk 3]
```diff
@@ -466,7 +465,6 @@ public void deleteFeedbackSessionsForCourses(List<String> courseIds) {
         return load()
                 .filter("sentPublishedEmail =", false)
                 .filter("isPublishedEmailEnabled =", true)
-                .filter("feedbackSessionType !=", FeedbackSessionType.PRIVATE)
```
Comment (whipermr5) [inline src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: `datastore-indexes.xml` can be updated.

**Comment by sukanta-27** [comment]
@whipermr5 One query. 

>set the start time to `TIME_REPRESENTS_NEVER`

why are we setting the start time as `TIME_REPRESENTS_NEVER`?

**Comment by whipermr5** [comment]
> why are we setting the start time as TIME_REPRESENTS_NEVER?

Was just using it as an example. Maybe set it to `createdTime` if it's `null` and leave it if it isn't?

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java (hunk 3 above)
Comment (sukanta-27) [reply src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: Can you clarify this a little. I didn't understand this. :sweat_smile:

Comment (whipermr5) [reply src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: This query is supported by a composite index (read more about Datastore indexes [here](https://cloud.google.com/datastore/docs/concepts/indexes)). Since we're updating the query, we should update the [index configuration](https://cloud.google.com/appengine/docs/standard/java/datastore/indexes#index-configuration) too.

**Comment by sukanta-27** [comment]
@whipermr5 Need some assistance. I wrote a method to convert `PRIVATE` sessions to `STANDARD` sessions, similar to what we are doing for startTime and end time. 

```java
    @OnLoad
    @SuppressWarnings("unused") // called by Objectify
    private void adjustFeedbackSessionType() {
        if (getFeedbackSessionType() == FeedbackSessionType.PRIVATE) {
            setFeedbackSessionType(FeedbackSessionType.STANDARD);
        }
    }
```
My question is how to migrate session type to `STANDARD` such that we can delete `PRIVATE` from `FeedbackSessionType.java`.

Currently deleting it causes error [screenshot below] if a course has any private session. (Sorry if this is trivial, Data migration is new to me :sweat_smile: 

screenshot 449

**Comment by whipermr5** [comment]
> My question is how to migrate session type to STANDARD such that we can delete PRIVATE from FeedbackSessionType.java.

If we still want to retain the field `feedbackSessionType` of type `FeedbackSessionType`, we'll have to keep the old definition of `FeedbackSessionType` there until we perform the actual migration. Alternatively, we can remove the field altogether.

**Comment by sukanta-27** [comment]
>If we still want to retain the field feedbackSessionType of type FeedbackSessionType, we'll have to keep the old definition of FeedbackSessionType there until we perform the actual migration

I think we should keep the field until the actual migration. will that be a better approach?

>Alternatively, we can remove the field altogether.

Actually at first I deleted the `PRIVATE` type, I understand that the existing session types will be changed after the actual migration, but how to permanently change the session types of the sessions in our local system? What am I missing here?

**Comment by whipermr5** [comment]
> Actually at first I deleted the PRIVATE type, I understand that the existing session types will be changed after the actual migration, but how to permanently change the session types of the sessions in our local system? What am I missing here?

Even the local system has existing data. If you were to wipe the database, your solution would work.

**Comment by sukanta-27** [comment]
>Even the local system has existing data. If you were to wipe the database, your solution would work.

I am aware of that. I have checked it. Removing the `PRIVATE` type works fine when there is no existing PRIVATE session. But wiping the database will not be a optimal solution right?

So  we should keep the Private type in `FeedbackSessionType` file till the actual migration happens right?

**Comment by whipermr5** [comment]
Yes. But actually that field has no more value after we remove private sessions.

**Comment by whipermr5** [comment]
Another way would be to use the `@IgnoreLoad` annotation.

**Comment by sukanta-27** [comment]
@whipermr5 
>Yes. But actually that field has no more value after we remove private sessions.

If by this you mean to remove the `FeedbackSessionType.java` altogether as we only have one session type right now, then I thought about that too at first. Then decided against it because of the `TODO`  mentioned in that file.

https://github.com/TEAMMATES/teammates/blob/8a974ee703c46f4b3280552f46b3e767eb561eab/src/main/java/teammates/common/datatransfer/FeedbackSessionType.java#L1-L8

If we in future decide to add more session types, then re-implementing the feedbackSessionType will be a lot of work. But of course the decision is for senior developers to make.

**Comment by whipermr5** [comment]
Yup I did see that. However, looking at the git history, nothing has changed in 5 years (other than removal of a `TEAM` type 4 years ago). If new types are ever added in future, it isn't that much work - Objectify makes it simple; we can always set a default value using `@OnLoad` for existing data.

**Comment by sukanta-27** [comment]
>Yup I did see that. However, looking at the git history, nothing has changed in 5 years (other than removal of a `TEAM`  type 4 years ago). If new types are ever added in future, it isn't that  much work - Objectify makes it simple; we can always set a default  value using `@OnLoad` for existing data.

I see. Then we can remove feedbackSessionType attribute altogether. working on that.

**Comment by sukanta-27** [comment]
@whipermr5 I have some confusion regarding feedback session type, so need to clarify before I proceed.

Does Teammates use `feedbackSessionType` to mean two different things? I mean while removing feedbackSessionType attribute, I was searching for its occurrences and found that `InstructorFeedbackAddAction`, `FeedbackSessionForm.java` and some files use the feedbackSessionType for referring to question templates. Like in this case,
https://github.com/TEAMMATES/teammates/blob/a88f169d8b34e1e50317ac91306aa015bc68b1ba/src/main/java/teammates/ui/controller/InstructorFeedbackAddAction.java#L125-L144

This method takes `templateType`, but when calling this method, we use the name `feedbackSessionType`
https://github.com/TEAMMATES/teammates/blob/a88f169d8b34e1e50317ac91306aa015bc68b1ba/src/main/java/teammates/ui/controller/InstructorFeedbackAddAction.java#L115-L117

And in the file `InstructorFeedbackSessionsPageData` the comment in line `L46` confuses me.
https://github.com/TEAMMATES/teammates/blob/a88f169d8b34e1e50317ac91306aa015bc68b1ba/src/main/java/teammates/ui/pagedata/InstructorFeedbackSessionsPageData.java#L45-L54

So, to be sure, when we are using the term `feedbackSessionType`, are we referring to question templates or session types? Or, are they both the same thing! Which will be surprising to me, as I thought session types and question templates are independent of each other. 


**P.S:** Sorry, if this comment is a bit unclear, if you face any doubt regarding what my doubt is, do let me know. :sweat_smile:

**Comment by whipermr5** [comment]
@sukanta-27 This may be leftover work from 4bbed734c0c7b9a21f741eeabc22db18804a57fd. It seems like this `feedbackSessionType` is only used to determine which template to use to populate the new session with questions, and not persisted to the datastore. We can rename all references to `templateType` to avoid confusion.

**Comment by sukanta-27** [comment]
Apologies as I couldn't focus on this for a couple of days as I was focusing on GSoC proposals. Continuing my work here.

**Comment by sukanta-27** [comment]
>We can rename all references to `templateType` to avoid confusion.

@whipermr5 How about we rename it to `sessionTemplateType` or `questionTemplateType` (I would prefer `sessionTemplateType`) as we also use templates in other places like emails. And using `templateType` may again create confusion?

**Comment by damithc** [comment]
I suppose the migration script will set the session visible times of existing private sessions to very far in the future (like 100 years)?

**Comment by sukanta-27** [comment]
@damithc Yes sir. The start time will be created time, and endTime will be 100 years into the future. (f395709f82d1e12cff7b69dff7e06ece83ed4ab7)

**Comment by damithc** [comment]
> @damithc Yes sir. The start time will be created time, and endTime will be 100 years into the future. (f395709)

I'm talking about 'visible from' setting, not start/end time. In any case, these sessions were not meant to be visible to others at all; shouldn't we set all times to be in 100 years in future to ensure they will not be visible to others in the course at all?

**Comment by sukanta-27** [comment]
>I'm talking about 'visible from' setting, not start/end time. In any case, these sessions were not meant to be visible to others at all; shouldn't we set all times to be in 100 years in future to ensure they will not be visible to others in the course at all?

@damithc Oops! I thought we were talking about start/end times. Yes, we should set the visible from time.

@whipermr5 In this case, how should I proceed. Should I write a method similar to what you suggested in `FeedbackSession.java` to set the `sessionVisibleFromTime` to be 100 years in the future so that #8448 can take care of the migration? (I will have to add the `FeedbackSessionType.java` to filter out the Private sessions for this, I deleted the file when removing the `feedbackSessionType` field).

```java
    @OnLoad
    @SuppressWarnings("unused") // called by Objectify
    private void adjustSessionVisibleTimeForPrivateSessions() {
        if (getFeedbackSessionType() == FeedbackSessionType.PRIVATE) {
            setSessionVisibleFromTime(TimeHelper.parseInstant("2118-01-01 12:00 AM +0000"))
        }
    }
```

**Comment by whipermr5** [comment]
No need to add the enum back; restore the field but change the field type to `String` and do a string comparison (use the datastore viewer `http://localhost:8080/_ah/admin` to see how the data is actually stored as a string all along). Can also annotate the field with `@IgnoreSave` so that the field will be deleted after conversion.

**Comment by sukanta-27** [comment]
@whipermr5 Ready for review.

File: src/main/java/teammates/storage/entity/FeedbackSession.java [hunk 4]
```diff
@@ -193,6 +192,25 @@ private void adjustResultsVisibleFromTimeIfRequired() {
...
+            setStartTime(getCreatedTime());
+        }
+        if (getEndTime() == null) {
+            setEndTime(TimeHelper.parseInstant("2118-01-01 12:00 AM +0000"));
+        }
+    }
+
+    @OnLoad
+    @SuppressWarnings("unused") // called by Objectify
+    private void adjustSessionVisibleTimeForPrivateSessions() {
```
Comment (whipermr5) [inline src/main/java/teammates/storage/entity/FeedbackSession.java]: Can combine this with the other method.

Comment (sukanta-27) [reply src/main/java/teammates/storage/entity/FeedbackSession.java]: Okay.

**Comment by sukanta-27** [comment]
@whipermr5 Should I remove the null checks in `isClosed()`, `isOpened()`, `isWaitingToOpen()` and `isInGracePeriod()` methods too?

**Comment by whipermr5** [comment]
@sukanta-27 Sounds good.

**Comment by sukanta-27** [comment]
@whipermr5 Changes made. Ready for review.

File: src/main/java/teammates/storage/entity/FeedbackSession.java [hunk 5]
```diff
@@ -193,6 +192,20 @@ private void adjustResultsVisibleFromTimeIfRequired() {
...
+    @SuppressWarnings("unused") // called by Objectify
+    private void adjustOpeningAndClosingAndVisibleTimeForPrivateSessions() {
+        if (getStartTime() == null) {
+            setStartTime(getCreatedTime());
+        }
+        if (getEndTime() == null) {
+            setEndTime(TimeHelper.parseInstant("2118-01-01 12:00 AM +0000"));
+        }
+        if ("PRIVATE".equals(feedbackSessionType)) {
+            setSessionVisibleFromTime(TimeHelper.parseInstant("2118-01-01 12:00 AM +0000"));
```
Comment (whipermr5) [inline src/main/java/teammates/storage/entity/FeedbackSession.java]: I realised that the session visible from time is never supposed to be after the session start time:
screen shot 2018-04-04 at 4 58 11 pm

Hence, it is the start time that we need to set to a hundred years later. The visible from time can be set to `TIME_REPRESENTS_FOLLOW_OPENING` so that it follows the start time.

**Comment by sukanta-27** [comment]
@whipermr5 Done. Ready for review.

**Comment by sukanta-27** [comment]
@whipermr5 I thought any test case that is for private sessions, will not need to get tested after the removal of private sessions :confused: Looking into it.

File: src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSessionsPageUiTest.java [hunk 6]
```diff
@@ -263,49 +259,6 @@ private void testAddAction() throws Exception {
                 newSession.getInstructions(), newSession.getGracePeriodMinutes());
         feedbackPage.waitForTextsForAllStatusMessagesToUserEquals(Const.StatusMessages.FEEDBACK_SESSION_EXISTS);
 
-        ______TS("success case: private session, boundary length name, only results email");
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSessionsPageUiTest.java]: E.g. boundary length name and only results email

**Comment by sukanta-27** [comment]
@whipermr5 I know this varies from test to test, but still want your advice on something. 
Should I look for test cases that are only focused on private sessions (I mean where there exists same test case for standard sessions too), and remove only those tests completely. And for all other private session test case, just remove the private session part and keep the other parts. will this be a good way to distinguish which tests to keep and which are not?

**Comment by whipermr5** [comment]
@sukanta-27 Yes, exactly! :+1::+1:

**Comment by sukanta-27** [comment]
@whipermr5 I don't think there is any other test that checks something that is not directly related to private sessions only. But still please have a look and notify me if you find any other such test. Also there are two tests that I have doubts about.

File: src/test/java/teammates/test/cases/browsertests/InstructorStudentRecordsPageUiTest.java [hunk 7]
```diff
@@ -63,18 +63,6 @@ private void testContent() throws Exception {
         viewPage = getStudentRecordsPage();
         viewPage.verifyHtmlMainContent("/instructorStudentRecordsWithHelperView.html");
 
-        ______TS("content: normal student records with other instructor's comments, private feedback session");
```
Comment (sukanta-27) [inline src/test/java/teammates/test/cases/browsertests/InstructorStudentRecordsPageUiTest.java]: I believe this is directly related to private sessions, and hence need to be deleted. right?

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java [hunk 8]
```diff
@@ -184,26 +184,6 @@ public void testGenerateFeedbackSessionEmails() throws IOException {
                                 session.getFeedbackSessionName());
         verifyEmail(email, instructor1.email, subject, "/sessionSubmissionConfirmationEmailZeroTimeZone.html");
 
-        ______TS("no email alerts sent for sessions not answerable/viewable for students");
```
Comment (sukanta-27) [inline src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: This test takes a private session to testing. But I am not sure if it is directly related to private sessions or not. (I assume it is). So, want you to confirm that this test needs deletion.

Comment (whipermr5) [reply src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: From reading the `TS` description, I believe needs to be a test case for sessions with no questions that are answerable for students. Can only remove this if such a test case exists elsewhere.

File: src/test/java/teammates/test/cases/browsertests/InstructorStudentRecordsPageUiTest.java (hunk 7 above)
Comment (whipermr5) [reply src/test/java/teammates/test/cases/browsertests/InstructorStudentRecordsPageUiTest.java]: What about the `normal student records with other instructor's comments` part? I'm guessing that the comments are not part of the private session?

Comment (sukanta-27) [reply src/test/java/teammates/test/cases/browsertests/InstructorStudentRecordsPageUiTest.java]: You were right. I misinterpreted this test.

File: src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java (hunk 8 above)
Comment (sukanta-27) [reply src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: Will have to add another session in the json file, and also blocked because of #8761

Comment (whipermr5) [reply src/test/java/teammates/test/cases/logic/EmailGeneratorTest.java]: No longer blocked :)

**Comment by sukanta-27** [comment]
@whipermr5 Changes made. Ready for review.

**Comment by sukanta-27** [comment]
@whipermr5 A gentle reminder. This is ready for review.

File: src/main/java/teammates/logic/core/FeedbackSessionsLogic.java [hunk 9]
```diff
@@ -2112,33 +2038,23 @@ private void addEmailNamePairsToTable(Map<String, String> emailNameTable,
...
-        List<FeedbackSessionDetailsBundle> fsDetailsWithoutPrivate = new ArrayList<>();
+        List<FeedbackSessionDetailsBundle> fsDetails = new ArrayList<>();
         List<FeedbackSessionAttributes> fsInCourse =
                 fsDb.getFeedbackSessionsForCourse(courseId);
 
         for (FeedbackSessionAttributes fsa : fsInCourse) {
-            if (!fsa.isPrivateSession() || fsa.isCreator(instructorEmail)) {
-                fsDetailsWithoutPrivate.add(getFeedbackSessionDetails(fsa));
+            if (fsa.isCreator(instructorEmail)) {
+                fsDetails.add(getFeedbackSessionDetails(fsa));
```
Comment (whipermr5) [inline src/main/java/teammates/logic/core/FeedbackSessionsLogic.java]: I believe the condition is wrong - the if is no longer necessary. The `isCreator` check was only applicable for private sessions.

File: src/main/java/teammates/ui/controller/InstructorFeedbackAbstractAction.java [hunk 10]
```diff
@@ -91,8 +89,7 @@ protected FeedbackSessionAttributes extractFeedbackSessionData(
             throw new InvalidPostParametersException("Invalid resultsVisibleFrom setting: " + type);
         }
 
-        // Handle session visible after results visible to avoid having a
-        // results visible date when session is private (session not visible)
+        // Handle session visible after results visible
```
Comment (whipermr5) [inline src/main/java/teammates/ui/controller/InstructorFeedbackAbstractAction.java]: The whole comment can be removed

File: src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java [hunk 11]
```diff
@@ -133,7 +133,7 @@ public void testExecuteAndPostProcess() {
 
         assertFalse(r.isError);
 
-        ______TS("Private case, empty participant list - data bundle params");
+        ______TS("Standard case, empty participant list - data bundle params");
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java]: Can just delete the whole case thing: `______TS("Empty participant list - data bundle params");`

File: src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java [hunk 12]
```diff
@@ -143,7 +143,7 @@ public void testExecuteAndPostProcess() {
         fs = typicalBundle.feedbackSessions.get("session1InCourse2");
         fq = FeedbackQuestionsLogic.inst().getFeedbackQuestion(fs.getFeedbackSessionName(), fs.getCourseId(), 1);
 
-        String[] privateParams = new String[] {
+        String[] standardParams = new String[] {
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java]: Use `params`

File: src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java [hunk 13]
```diff
@@ -159,15 +159,14 @@ public void testExecuteAndPostProcess() {
...
         };
-
-        a = getAction(privateParams);
+        a = getAction(standardParams);
         r = getAjaxResult(a);
 
         assertFalse(r.isError);
 
-        ______TS("Private case, null participant list - constructed params");
+        ______TS("Standard case, null participant list - constructed params");
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java]: Same here.

File: src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java [hunk 14]
```diff
@@ -159,15 +159,14 @@ public void testExecuteAndPostProcess() {
...
 
-        privateParams = new String[] {
+        standardParams = new String[] {
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java]: Same here.

File: src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java [hunk 15]
```diff
@@ -184,10 +183,11 @@ public void testExecuteAndPostProcess() {
                 Const.ParamsNames.FEEDBACK_QUESTION_ID, fq.getId()
         };
 
-        a = getAction(privateParams);
+        a = getAction(standardParams);
         r = getAjaxResult(a);
 
         assertFalse(r.isError);
+
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/action/InstructorFeedbackQuestionVisibilityMessageActionTest.java]: Unrelated change.

File: src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSessionsPageUiTest.java [hunk 16]
```diff
@@ -50,15 +49,14 @@ protected void prepareTestData() {
                 .builder("New Session ##", course.getId(), "teammates.test1@gmail.tmt")
                 .withStartTime(TimeHelper.parseInstant("2035-04-01 9:59 PM +0000"))
                 .withEndTime(TimeHelper.parseInstant("2035-04-30 8:00 PM +0000"))
-                .withCreatedTime(Const.TIME_REPRESENTS_NEVER)
+                .withCreatedTime(Const.TIME_REPRESENTS_NOW)
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSessionsPageUiTest.java]: Was this needed to fix the test? If so, would an arbitrary timestamp be better than the special timestamp (I don't think the special one is possible in real usage)?

File: src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSessionsPageUiTest.java [hunk 17]
```diff
@@ -273,24 +269,19 @@ private void testAddAction() throws Exception {
...
         feedbackPage.clickEditUncommonSettingsButtons();
-        feedbackPage.clickNeverVisibleTimeButton();
 
-        //verify that timeFrameTable, instructions and ResponseVisTable are all hidden
-        assertTrue(feedbackPage.isHidden(By.id("timeFramePanel")));
-        assertTrue(feedbackPage.isHidden(By.id("responsesVisibleFromColumn")));
-        assertTrue(feedbackPage.isHidden(By.id("instructionsRow")));
-
-        newSession.setFeedbackSessionName("private session of characters1234567 #");
+        newSession.setFeedbackSessionName("standard session of characters12345678");
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/browsertests/InstructorFeedbackSessionsPageUiTest.java]: Remove standard. It hints that there used to be another type of session other than standard.

File: src/test/java/teammates/test/cases/logic/FeedbackSessionsLogicTest.java [hunk 18]
```diff
@@ -570,26 +463,12 @@ private void testGetFeedbackSessionsForCourse() throws Exception {
...
-
-        ______TS("Private session viewing");
-
-        // This is the creator for the private session.
-        // We have already tested above that other instructors cannot see it.
-        actualSessions = fsLogic.getFeedbackSessionsForUserInCourse("idOfTypicalCourse2", "instructor1@course2.tmt");
-        AssertHelper.assertContains(dataBundle.feedbackSessions.get("session1InCourse2").toString(),
-                actualSessions.toString());
-
-        ______TS("Feedback session without questions for students but with visible responses are visible");
```
Comment (whipermr5) [inline src/test/java/teammates/test/cases/logic/FeedbackSessionsLogicTest.java]: Was this TS related to private sessions?

**Comment by sukanta-27** [comment]
@whipermr5 Changes made. Ready for review.

**Comment by whipermr5** [comment]
@damithc Since the session visible time for existing private sessions will now be in the future, instructors will no longer be able to submit responses. I believe this is okay? They will still be able to view results of their previously-submitted responses.

**Comment by whipermr5** [comment]
> They will still be able to view results of their previously-submitted responses.

I just realised this means that other instructors will now be able to see their private responses - this might be bad?

**Comment by sukanta-27** [comment]
@whipermr5 I guess from a previous https://github.com/TEAMMATES/teammates/pull/8408#issuecomment-364850538 by Prof. @damithc that this is expected?

**Comment by damithc** [comment]
> @damithc Since the session visible time for existing private sessions will now be in the future, instructors will no longer be able to submit responses. I believe this is okay? They will still be able to view results of their previously-submitted responses.

Yes it is fine. This is the what I expect to happen.

1. All dates of current private sessions are automatically pushed to a date far in future. Until that date is reached, others should not be able to see those sessions. This is important as those sessions were meant to be private.
2. Owner of the session can no longer submit responses to it. That's fine because we no longer support the concept of private sessions.
3. We inform owners of private sessions that the feature is no longer supported and ask them do do whatever they see fit with the privates sessions. e.g., delete them

Is the current state of this PR compatible with the above workflow?

**Comment by damithc** [comment]
> I believe this part is not - all instructors are able to see all sessions in the course, even if the sessions are not visible/open to students yet.

I see. In that case, we need to inform those instructors before we deploy this PR.

**Comment by sukanta-27** [comment]
@wkurniawan07 A gentle reminder. This is ready for review. :sweat_smile:

**Comment by damithc** [comment]
Note: I have notified relevant instructors that we are removing private sessions. This PR can be included in the next release.

**Comment by sukanta-27** [comment]
>2.5K fewer LOC to maintain! Good work @sukanta-27

Thank you very much sir. :smile:
